### PR TITLE
elif clang-min definition changes

### LINKED ISSUE #25991: Create a min-clang container
## Reason
We want to support our Minimum Clang target using an Apptainer.

## Design
I am thinking a `ALTERNATE_FROM = clang-min` modification to mpich.def.

## Impact
Version control the Minimum Clang def file, as we do with Modern Clang.

### PR DESCRIPTION
**Pull request by milljm**
Build our minimum version of supported Clang

Closes #25991

### CODE REVIEW

**Comment by milljm** [comment]
Initial go... works manually. Need to create "allow_on_pr_num" test recipes...

**Comment by loganharbour** [comment]
While we're here, we should put this in post:

```
export MOOSE_JOBS={{ MOOSE_JOBS or "1" }}
```

replace the `${MOOSE_JOBS:-6}` with just `${MOOSE_JOBS}` and add

```
{#- MOOSE_JOBS: Number of jobs to pass to the build                                           -#}
```

to the jinja comment at the top

**Comment by milljm** [comment]
@loganharbour if you happen to be online, and think this is ready...

**Comment by loganharbour** [comment]
Do we need anything else done since this is another full rebuild?

**Comment by milljm** [comment]
Probably this: https://github.com/idaholab/moose/pull/26023

File: apptainer/mpich.def [hunk 1]
```diff
@@ -115,45 +148,75 @@ EOF
...
       --enable-fast=O2 \
       --enable-debuginfo \
-      --enable-totalview \
       --enable-two-level-namespace \
       CC=clang \
       CXX=clang++ \
       FC=gfortran \
       F77=gfortran \
-      F90='' \
+      F90=gfortan \
```
Comment (cticenhour) [inline apptainer/mpich.def:197]:
```suggestion
      F90=gfortran \
```

Found a typo

**Review by cticenhour — CHANGES_REQUESTED** [review]

File: apptainer/mpich.def (hunk 1 above)
Comment (milljm) [reply apptainer/mpich.def:197]: wow, how did this pas!?

**Comment by milljm** [comment]
Closing in favor of #26023

**Pull request closed without merging**